haystack/management/commands/update_index.py

```diff
@@ -86,14 +87,35 @@ def do_update(backend, index, qs, start, end, total, verbosity=1, commit=True):
     small_cache_qs = qs.all()
     current_qs = small_cache_qs[start:end]
 
+    is_parent_process = hasattr(os, 'getppid') and os.getpid() == os.getppid()
     if verbosity >= 2:
-        if hasattr(os, 'getppid') and os.getpid() == os.getppid():
+        if is_parent_process:
             print("  indexed %s - %d of %d." % (start + 1, end, total))
         else:
             print("  indexed %s - %d of %d (by %s)." % (start + 1, end, total, os.getpid()))
 
-    # FIXME: Get the right backend.
-    backend.update(index, current_qs, commit=commit)
+    max_retries = 5
+    retries = 0
+    while retries < max_retries:
+        try:
+            # FIXME: Get the right backend.
+            backend.update(index, current_qs, commit=commit)
+            if verbosity >= 2 and retries:
+                print('Succeeded indexing {} - {}, tried {}/{} times'.format(start + 1, end, retries + 1, max_retries))
+            break
+        except:
+            retries += 1
+            if verbosity >= 2:
+                exc = traceback.format_exc()
```

Comment:
I would prefer to use `logging.exception` (or `logging.warning` with `exc_info=True`) for this since it avoids the need for custom formatting logic, sending errors to `stderr` instead of `stdout`, etc. 

In this case I'd also like to avoid the blanket except in favor of catching a specific category of exception (e.g. `EnvironmentError`, `IOError`, etc.) so things like `KeyboardInterrupt` and `SystemExit` still work.